### PR TITLE
docs: Add forward_proxy_url parameter introduction

### PR DESCRIPTION
**Pull request by cismous**
feat: https://github.com/caddyserver/caddy/pull/6114

### CODE REVIEW

File: src/docs/markdown/caddyfile/directives/reverse_proxy.md [hunk 1]
```diff
@@ -593,6 +594,11 @@ transport http {
 
 - **max_conns_per_host** <span id="max_conns_per_host"/> optionally limits the total number of connections per host, including connections in the dialing, active, and idle states. Default: No limit.
 
+- **forward_proxy_url** <span id="forward_proxy_url"/> is a parameter that specifies the URL of a server that the HTTP transport will use to proxy requests to the upstream server. This parameter takes precedence over environment variables like HTTP_PROXY. When a value is provided for this parameter, requests will flow through the reverse proxy in the following order:
+  a. User Agent -> Reverse Proxy
+  b. Reverse Proxy -> Forward Proxy ( specified by forward_proxy_url )
```
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/reverse_proxy.md]:
```suggestion
  b. Reverse Proxy -> Forward Proxy (specified by `forward_proxy_url`)
```

**Review by mholt — APPROVED** [review]
LGTM! Thank you :)

**Comment by cismous** [comment]
@mholt  Hi, Why wasn't it merged? Is there anything else that needs to be done?

**Comment by mholt** [comment]
I forgot. Thank you!

**Comment by francislavoie** [comment]
Uh, it wasn't merged because the feature isn't released yet. @mholt we shouldn't have merged this.

**Pull request merged**